src/commonMain/kotlin/com/configcat/ConfigCatClient.kt

```diff
@@ -254,7 +254,15 @@ public interface ConfigCatClient {
 public fun ConfigCatClient(
     sdkKey: String,
     block: ConfigCatOptions.() -> Unit = {},
-): ConfigCatClient = Client.get(sdkKey, block)
+): ConfigCatClient = Client.get(sdkKey, ConfigCatOptions().apply(block))
+
+/**
+ * Creates a new or gets an already existing [ConfigCatClient] for the given [sdkKey].
+ */
+public fun ConfigCatClient(
+    sdkKey: String,
+    options: ConfigCatOptions,
+): ConfigCatClient = Client.get(sdkKey, options)
```

Comment:
What is the reason behind adding this new constructor? Is this necessary for tests maybe?

Anyway, it doesn't seem to be a good idea to provide two constructors that achieve exactly the same because it can be confusing to the consumers ("Which one should I use? What is the difference?")

If the new one is really necessary, probably we should deprecate the old one.

Reply:
It just allows a different construction style. It's useful when the component that initializes a client gets an already existing `ConfigCatOptions`. I don't want to deprecate the old one, as it's still valid for in-place construction.

I can add some examples to the API docs that could help users decide about which one to use.